src/features/SwapAssets/components/SwapAssets.tsx

```diff
@@ -89,10 +91,12 @@ const SwapAssets = (props: Props) => {
     isEstimatingGas,
     expiration,
     isDemoMode,
-    baseAsset,
     settings
   } = props;
 
+  const network = useSelector(selectNetwork(selectedNetwork)) as Network;
+  const baseAsset = useSelector(getBaseAssetByNetwork(network)) as ExtendedAsset;
```

Comment:
Why these type casts? It should be able to infer the types from the selector?

Reply:
It is not able to infer it when using functions that create selectors it seems.